File: examples/run-actor.js
// Node.js (CommonJS) example: run the Maritime Sanctions Monitor via the apify-client SDK.
// Install: npm install apify-client
// Usage:   APIFY_TOKEN=your_token node examples/run-actor.js

const { ApifyClient } = require('apify-client');

const client = new ApifyClient({
    token: process.env.APIFY_TOKEN,
});

async function main() {
    const run = await client.actor('dR68wHyuOLS2WEhmo').call({
        maxItems: 100,
        onlyNew: true,
        enrichWithUnConsolidatedList: true,
        programFilter: ['IRAN', 'RUSSIA-EO14024'],
    });

    const { items } = await client.dataset(run.defaultDatasetId).listItems();

    for (const item of items) {
        console.log(
            `${item.vesselName} (IMO ${item.imoNumber ?? 'n/a'}) - ${item.event_type} - programs: ${(item.programs || []).join(', ')}`,
        );
    }

    console.log(`Total vessel records: ${items.length}`);
}

main().catch((err) => {
    console.error('Actor run failed:', err);
    process.exit(1);
});

File: examples/run_actor.py
# Python example: run the Maritime Sanctions Monitor via the apify-client SDK.
# Install: pip install apify-client
# Usage:   APIFY_TOKEN=your_token python examples/run_actor.py

import os

from apify_client import ApifyClient

client = ApifyClient(token=os.environ["APIFY_TOKEN"])

run_input = {
    "maxItems": 100,
    "onlyNew": True,
    "enrichWithUnConsolidatedList": True,
    "programFilter": ["IRAN", "RUSSIA-EO14024"],
}

run = client.actor("dR68wHyuOLS2WEhmo").call(run_input=run_input)

items = client.dataset(run["defaultDatasetId"]).list_items().items

for item in items:
    imo = item.get("imoNumber") or "n/a"
    programs = ", ".join(item.get("programs", []))
    print(f"{item['vesselName']} (IMO {imo}) - {item['event_type']} - programs: {programs}")

print(f"Total vessel records: {len(items)}")
